### PR TITLE
Readme

### PR DESCRIPTION
**Pull request by mayabrandi**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,110 @@
...
+
+## Front End
+All views in vogue should be self explanatory. There should be no further documentation needed to be able to be interpret the content of the web page.
+
+## Back End
+The trending database is a Mongo database consisting of following collections:
+
+- **sample** - holds lims specific data on sample level. keys are lims sample ids
+- **sample_analysis** - holds data from diferent pipeliens on sample level. keys are lims sample ids
+- **case_analysis** - holds data from diferent pipeliens on case level. keys are ???
```
Comment (henrikstranneheim) [inline README.md]: If there are none - remove it for now and let's add it later when appropriate

File: README.md [hunk 2]
```diff
@@ -1 +1,110 @@
...
+pip install -e .
+```
+
+## Front End
+All views in vogue should be self explanatory. There should be no further documentation needed to be able to be interpret the content of the web page.
+
+## Back End
+The trending database is a Mongo database consisting of following collections:
+
+- **sample** - holds lims specific data on sample level. keys are lims sample ids
```
Comment (henrikstranneheim) [inline README.md]: Do you mean anchoring identifier are LIMS sample ids?
Always use uppercase for acronyms: lims vs LIMS

File: README.md [hunk 3]
```diff
@@ -1 +1,110 @@
...
+
+The load of each collection is described below.
+
+## Data Flow
+<p align="center">
+		<img src="artwork/data_flow.png">
+</p>
+
+## CLI
+The cli has two base commands - load and run. The load is for loading various data into the trending database, and the run is for running the web application.
```
Comment (henrikstranneheim) [inline README.md]: The load -> The load command
the run -> the run command

File: README.md [hunk 4]
```diff
@@ -1 +1,110 @@
...
+
+### load sample
```
Comment (henrikstranneheim) [inline README.md]: load sample -> Load

File: README.md [hunk 5]
```diff
@@ -1 +1,110 @@
 # vogue (**version** = 0.2.0)
+
+Vogue is clinical genomics package for trending all kinds of data...
```
Comment (hassanfa) [inline README.md]: is a Clinical Genomics

File: README.md [hunk 6]
```diff
@@ -1 +1,110 @@
...
+</p>
+
+## CLI
+The cli has two base commands - load and run. The load is for loading various data into the trending database, and the run is for running the web application.
+
+### load sample
+```
+Usage: vogue load sample [OPTIONS]
+
+  Read and load lims data for one ore all samples. When loading many
```
Comment (henrikstranneheim) [inline README.md]: ore -> or

File: README.md [hunk 7]
```diff
@@ -1 +1,110 @@
...
+  smaples, the different options -f, -n, -d are used to delimit the subset
```
Comment (henrikstranneheim) [inline README.md]: smaples -> samples

File: README.md [hunk 8]
```diff
@@ -1 +1,110 @@
...
+
+
+```bash
+git clone https://github.com/Clinical-Genomics/vogue.git
+cd vogue
+pip install -e .
+```
+
+## Front End
+All views in vogue should be self explanatory. There should be no further documentation needed to be able to be interpret the content of the web page.
```
Comment (hassanfa) [inline README.md]: self-explanatory

File: README.md [hunk 9]
```diff
@@ -1 +1,110 @@
...
+
+Options:
+  -s, --sample-lims-id TEXT  Input sample lims id
+  -m, --many                 Loads all lims samples if no other options are
+                             selected
+  --dry                      Load from sample or not. (dry-run)
+  -f, --load-from TEXT       load from this sample lims id. Use if load all
+                             broke. Start where it ended
+  -n, --new                  Use this flagg if you only want to load samples
+                             that dont exist in the database
```
Comment (henrikstranneheim) [inline README.md]: dont -> do not

File: README.md [hunk 10]
```diff
@@ -1 +1,110 @@
...
+  Read and load lims data for one ore all samples. When loading many
+  smaples, the different options -f, -n, -d are used to delimit the subset
+  of samples to load.
+
+Options:
+  -s, --sample-lims-id TEXT  Input sample lims id
+  -m, --many                 Loads all lims samples if no other options are
+                             selected
+  --dry                      Load from sample or not. (dry-run)
+  -f, --load-from TEXT       load from this sample lims id. Use if load all
```
Comment (henrikstranneheim) [inline README.md]: load -> Load

File: README.md (hunk 1 above)
Comment (hassanfa) [inline README.md]: Cases are case names (the pet names). But I am not sure about this.

File: README.md [hunk 11]
```diff
@@ -1 +1,110 @@
...
+                             broke. Start where it ended
+  -n, --new                  Use this flagg if you only want to load samples
+                             that dont exist in the database
+  -d, --date TEXT            Update only samples delivered after date
+  --help                     Show this message and exit.
+  ```
+  
+  ### load analysis
```
Comment (henrikstranneheim) [inline README.md]: load -> Load

File: README.md [hunk 12]
```diff
@@ -1 +1,110 @@
...
+  ```
+  Usage: vogue load analysis [OPTIONS]
+
+  Read and load analysis results. These are either QC or analysis output
+  files.
+
+  The inputs are unique ID with an analysis config file (JSON/YAML) which
```
Comment (henrikstranneheim) [inline README.md]: The inputs -> The input are unique IDs

File: README.md [hunk 13]
```diff
@@ -1 +1,110 @@
...
+  includes analysis results matching the analysis model. Analysis types
```
Comment (henrikstranneheim) [inline README.md]: Do you mean?
Supported analysis types keys are: QC:multiqc_picard_dup,
  multiqc_picard_HsMetrics, multiqc_picard_AlignmentSummaryMetrics,
  multiqc_picard_insertSize

File: README.md [hunk 14]
```diff
@@ -1 +1,110 @@
...
+Options:
+  -s, --sample-id TEXT          Input sample id  [required]
+  -a, --analysis-config PATH    Input config file. Accepted format: JSON, YAML
+                                [required]
+  -t, --analysis-type [QC|all]  Type of analysis results to load.
+  --dry                         Load from sample or not. (dry-run)
+  --help                        Show this message and exit.
+  ```
+  
+  ### load flowcell
```
Comment (henrikstranneheim) [inline README.md]: Load

File: README.md [hunk 15]
```diff
@@ -1 +1,110 @@
...
+  Read and load lims data for one or all runs
+
+Options:
+  -r, --run-id TEXT  Lims process id for the run. Eg: 24-100451
+  -a, --all-runs     Loads all lims flowcells ids
+  --dry              Load from flowcell or not. (dry-run)
+  --help             Show this message and exit.
+  ```
+  
+  ### load apptag
```
Comment (henrikstranneheim) [inline README.md]: Load

File: README.md [hunk 16]
```diff
@@ -1 +1,110 @@
...
+  Usage: vogue load apptag [OPTIONS] APPLICATION_TAGS
+
+  Reads json string with application tags. Eg:'[{"tag":"MELPCFR030",
+  "category":"wgs",...},...]'
+
+Options:
+  --help  Show this message and exit.
+  ```
+  
+### run
```
Comment (henrikstranneheim) [inline README.md]: Run

File: README.md [hunk 17]
```diff
@@ -1 +1,107 @@
...
+
+
+```bash
+git clone https://github.com/Clinical-Genomics/vogue.git
+cd vogue
+pip install -e .
+```
+
+## Front End
+All views in vogue should be self explanatory. There should be no further documentation needed to be able to be interpret the content of the web page.
```
Comment (ingkebil) [inline README.md]:
```suggestion
All views in vogue should be self explanatory. There should be no further documentation needed to be able to interpret the content of the web page.
```

File: README.md [hunk 18]
```diff
@@ -1 +1,107 @@
...
+
+## Back End
+The trending database is a Mongo database consisting of following collections:
+
+- **sample** - holds lims specific data on sample level. keys are lims sample ids
+- **sample_analysis** - holds data from diferent pipeliens on sample level. keys are lims sample ids
```
Comment (ingkebil) [inline README.md]:
```suggestion
- **sample_analysis** - holds data from different pipelines on sample level. Keys are lims sample ids
```

File: README.md [hunk 19]
```diff
@@ -1 +1,107 @@
...
+- **case_analysis** - holds data from diferent pipeliens on case level. keys are ???
+- **flowcell** - holds lims specific data on run level. keys are flowcell ids
```
Comment (ingkebil) [inline README.md]:
```suggestion
- **flowcell** - holds lims specific data on run level. Keys are flowcell ids
```

File: README.md [hunk 20]
```diff
@@ -1 +1,107 @@
...
+- **application_tag** - holds application tag specific data. Keys are application tags.
+
+The load of each collection is described below.
```
Comment (ingkebil) [inline README.md]:
```suggestion
Loading of each collection is described below.
```

File: README.md [hunk 21]
```diff
@@ -1 +1,107 @@
...
+
+## Data Flow
+<p align="center">
+		<img src="artwork/data_flow.png">
+</p>
+
+## CLI
+The cli has two base commands - load and run. The load is for loading various data into the trending database, and the run is for running the web application.
```
Comment (ingkebil) [inline README.md]:
```suggestion
The CLI has two base commands - load and run. The load command is for loading various data into the trending database, and the run command is for running the web application.
```

File: README.md [hunk 22]
```diff
@@ -1 +1,107 @@
...
+
+  Read and load lims data for one ore all samples. When loading many
+  smaples, the different options -f, -n, -d are used to delimit the subset
+  of samples to load.
+
+Options:
+  -s, --sample-lims-id TEXT  Input sample lims id
+  -m, --many                 Loads all lims samples if no other options are
+                             selected
+  --dry                      Load from sample or not. (dry-run)
```
Comment (ingkebil) [inline README.md]: Could we use `--dry-run` instead of `--dry` to be consistent with other apps?

File: README.md [hunk 23]
```diff
@@ -1 +1,107 @@
...
+  -f, --load-from TEXT       load from this sample lims id. Use if load all
+                             broke. Start where it ended
+  -n, --new                  Use this flagg if you only want to load samples
```
Comment (ingkebil) [inline README.md]:
```suggestion
  -n, --new                  Use this flag if you only want to load samples
```

File: README.md [hunk 24]
```diff
@@ -1 +1,107 @@
...
+  ### load flowcell
+  
+  ```
+  Usage: vogue load flowcell [OPTIONS]
+
+  Read and load lims data for one or all runs
+
+Options:
+  -r, --run-id TEXT  Lims process id for the run. Eg: 24-100451
+  -a, --all-runs     Loads all lims flowcells ids
```
Comment (ingkebil) [inline README.md]:
```suggestion
  -a, --all-runs     Load all lims flowcells ids
```

File: README.md [hunk 25]
```diff
@@ -1 +1,107 @@
...
+  
+  ### load flowcell
+  
+  ```
+  Usage: vogue load flowcell [OPTIONS]
+
+  Read and load lims data for one or all runs
+
+Options:
+  -r, --run-id TEXT  Lims process id for the run. Eg: 24-100451
```
Comment (ingkebil) [inline README.md]:
```suggestion
  -r, --run-id TEXT  LIMS process id for the run. Eg: 24-100451
```

File: README.md (hunk 13 above)
Comment (mayabrandi) [reply README.md]: @hassanfa !?

File: README.md (hunk 12 above)
Comment (mayabrandi) [reply README.md]: 👍

File: README.md (hunk 6 above)
Comment (mayabrandi) [reply README.md]: 👍

File: README.md (hunk 4 above)
Comment (mayabrandi) [reply README.md]: I cant change the header to just Load here. Each header from here and down, are the diferent load commands..  load sample, load analysis, load flowcell, etc. Would it be more clear if the header is more explicit? Load Sample Collection, Load Analysis Collection, Load Flowcell Collection, etc?

File: README.md (hunk 3 above)
Comment (mayabrandi) [reply README.md]: 👍

File: README.md (hunk 2 above)
Comment (mayabrandi) [reply README.md]: Didn't know about the word "anchoring identifier". But it sounds good.  :)

File: README.md (hunk 1 above)
Comment (mayabrandi) [reply README.md]: 👍

Comment (mayabrandi) [reply README.md]: Ill remove it for now

File: README.md (hunk 8 above)
Comment (mayabrandi) [reply README.md]: 👍

File: README.md (hunk 14 above)
Comment (mayabrandi) [reply README.md]: Ahaaa. Now I understand what you mean. :)

File: README.md (hunk 16 above)
Comment (mayabrandi) [reply README.md]: 👍

File: README.md (hunk 17 above)
Comment (mayabrandi) [reply README.md]: 👍

File: README.md (hunk 21 above)
Comment (mayabrandi) [reply README.md]: 👍

File: README.md [hunk 26]
```diff
@@ -12,88 +12,98 @@ pip install -e .
...
+```
+Usage: vogue load flowcell [OPTIONS]
 
-  Read and load lims data for one or all runs
+  Read and load LIMS data for one or all runs
 
 Options:
-  -r, --run-id TEXT  Lims process id for the run. Eg: 24-100451
-  -a, --all-runs     Loads all lims flowcells ids
+  -r, --run-id TEXT  Flowcell id. Eg: 190315_D00410_0873_BHWWKCBCX2
```
Comment (ingkebil) [inline README.md]: In other contexts, a flowcell id is only  HWWKCBCX2. Please note that that excludes the `B` which denotes the flowcell position on the sequencer.

File: vogue/commands/load/flowcell.py [hunk 27]
```diff
@@ -10,9 +10,9 @@
 
 LOG = logging.getLogger(__name__)
 
-@click.command("flowcell", short_help = "load flowcell into db.")
-@click.option('-r', '--run-id', help = 'Lims process id for the run. Eg: 24-100451')
-@click.option('-a', '--all-runs', is_flag = True, help = 'Loads all lims flowcells ids')
+@click.command("flowcell", short_help = "Load flowcell into db.")
+@click.option('-r', '--run-id', help = 'Flowcell id. Eg: 190315_D00410_0873_BHWWKCBCX2')
```
Comment (ingkebil) [inline vogue/commands/load/flowcell.py]: Same here: a flowcell id is HWWKCBCX2

**Comment by ingkebil** [comment]
Well written :+1:

File: vogue/commands/load/flowcell.py (hunk 27 above)
Comment (mayabrandi) [reply vogue/commands/load/flowcell.py]: I changed back to lims process id witch is more correct in this case. We want to be able to load data from a specific process with this option.

Comment (ingkebil) [reply vogue/commands/load/flowcell.py]: Can you then change the help field to reflect that? e.g. 
```suggestion
@click.option('-r', '--run-id', help = 'Run id. Eg: 190315_D00410_0873_BHWWKCBCX2')
```

Comment (mayabrandi) [reply vogue/commands/load/flowcell.py]: Im quite sure I did @ingkebil

**Pull request merged**